test/utils/IntegrationTest.sol

```diff
@@ -0,0 +1,415 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity 0.8.20;
+
+import { ERC20ForwarderFactory } from "contracts/src/token/ERC20ForwarderFactory.sol";
+import { HyperdriveFactory } from "contracts/src/factory/HyperdriveFactory.sol";
+import { IERC20 } from "contracts/src/interfaces/IERC20.sol";
+import { IHyperdrive } from "contracts/src/interfaces/IHyperdrive.sol";
+import { AssetId } from "contracts/src/libraries/AssetId.sol";
+import { ETH } from "contracts/src/libraries/Constants.sol";
+import { FixedPointMath, ONE } from "contracts/src/libraries/FixedPointMath.sol";
+import { HyperdriveTest } from "test/utils/HyperdriveTest.sol";
+import { HyperdriveUtils } from "test/utils/HyperdriveUtils.sol";
+import { Lib } from "test/utils/Lib.sol";
+
+/// @author DELV
+/// @title IntegrationTest
+/// @notice The base contract for the integration testing suite.
+/// @dev A testing suite that provides a foundation to setup, deploy, and
+/// test common cases for Hyperdrive instances.
+/// @custom:disclaimer The language used in this code is for coding convenience
+///                    only, and is not intended to, and does not, have any
+///                    particular legal or regulatory significance.
+abstract contract IntegrationTest is HyperdriveTest {
+    using Lib for *;
+    using FixedPointMath for uint256;
+
+    // Fixed rate used to configure market.
+    uint256 internal constant FIXED_RATE = 0.05e18;
+
+    // ETH whale account.
+    address internal ETH_WHALE = 0x00000000219ab540356cBB839Cbe05303d7705Fa;
+
+    /// @dev Configuration for the Integration testing suite.
+    struct IntegrationConfig {
+        address[] whaleAccounts;
+        IERC20 token;
+        IERC20 baseToken;
+        uint256 shareTolerance;
+        uint256 minTransactionAmount;
+        uint256 positionDuration;
+        bool enableBaseDeposits;
+    }
+
+    IntegrationConfig internal config;
+    IHyperdrive.PoolDeployConfig internal poolConfig;
+    HyperdriveFactory factory;
+    address deployerCoordinator;
+    bool internal immutable isBaseETH;
```

Comment:
Let's document these. Should the `poolConfig` be passed in the constructor? I see it referenced below, but it isn't defined in this file. Furthermore, anything that isn't supposed to be accessed outside of this file should be marked `private`.

Reply:
Ok, I see that the pool config is set using the factory's configuration. Let's make it private in that case.